tests/sync/app.cpp

```diff
@@ -1908,6 +1908,67 @@ TEST_CASE("app: sync integration", "[sync][app]") {
         util::EventLoop::main().run_until([&]{ return error_did_occur.load(); });
         REQUIRE(error_did_occur.load());
     }
+
+    SECTION("invalid pk schema error handling")
+    {
+        TestSyncManager sync_manager({.app_config = app_config});
+        auto app = get_app_and_login(sync_manager.app());
+
+        const auto dog_schema = realm::ObjectSchema("Dog", {
+            realm::Property("my_primary_key", PropertyType::ObjectId | PropertyType::Nullable, true),
+            realm::Property("breed", PropertyType::String | PropertyType::Nullable),
+            realm::Property("name", PropertyType::String),
+            realm::Property("realm_id", PropertyType::String | PropertyType::Nullable)
+        });
+        const auto person_schema = realm::ObjectSchema("Person", {
+            realm::Property("_id", PropertyType::ObjectId | PropertyType::Nullable, true),
+            realm::Property("age", PropertyType::Int),
+            realm::Property("dogs", PropertyType::Object | PropertyType::Array, "Dog"),
+            realm::Property("firstName", PropertyType::String),
+            realm::Property("lastName", PropertyType::String),
+            realm::Property("realm_id", PropertyType::String | PropertyType::Nullable)
+        });
+        Schema schema = {dog_schema, person_schema};
+        auto config = setup_and_get_config(app, schema);
+        std::string exception_message;
+        try {
+            realm::Realm::get_shared_realm(config);
+        }
+        catch (const std::exception& e) {
+            exception_message = e.what();
+        }
+        CHECK(exception_message.find("The primary key property on a synchronized Realm must be named '_id' but found 'my_primary_key' for type 'Dog'") != std::string::npos);
```

Comment:
This is `REQUIRE_THROWS_WITH`

Reply:
There's a prefix to the exception message that I didn't want to check for, but I found `REQUIRE_THROWS_CONTAINING` elsewhere in the code so I moved to using that.